.pre-commit-config.yaml

```diff
@@ -12,16 +12,20 @@ repos:
   - id: check-yaml
   - id: check-executables-have-shebangs
 - repo: https://github.com/macisamuele/language-formatters-pre-commit-hooks
-  rev: v1.2.3
+  rev: v1.3.2
   hooks:
   - id: pretty-format-java
     args: [--autofix]
   - id: pretty-format-kotlin
     args: [--autofix]
   - id: pretty-format-yaml
     args: [--autofix, --indent, '2']
+    exclude: ^config/detekt/detekt\.yml$
 - repo: https://github.com/Yelp/detect-secrets
-  rev: v0.12.0
+  rev: v0.13.0
   hooks:
   - id: detect-secrets
     args: [--baseline, .secrets.baseline]
+
+# Do NOT run pre-commit hooks on generated-code
+exclude: ^samples/[^/]+/src/main/java
```

Comment:
Are we sure about this?
KtLint was sanitising the generated code a lot. Now it doesn't really look that elegant as before. Any arguments for having this `exclude` rule here?

Reply:
The proposal here, and I'm more than happy to remove this line if needed, is that `samples` should contain the exact output of the tool. 

If the generated code does not look "elegant" then we should eventually update the tool to generate some more elegant result rather than having ktlint fixing it.

One of the many things that I was noticing is that we have `import okhttp3.RequestBody` on all *Api.kt files even if we don't have parameters of `type: file`.

Reply:
> If the generated code does not look "elegant" then we should eventually update the tool to generate some more elegant result rather than having ktlint fixing it.

This is not always possible due to limitations of mustache. 
I'm generally fine with:

generator outputting not elegant code -> formatter fixing it -> eventually committing it.

Reply:
What might eventually be a concern is that having the formatter fixing the file might hide issues within the generated code.

Think about the [`import okhttp3.RequestBody`](https://github.com/Yelp/swagger-gradle-codegen/blob/9d9edcede7e2c719f9e41cec2ae9609444cd92ae/plugin/src/main/resources/kotlin/retrofit2/api.mustache#L4). The linter removes it because `RequestBody` is not used in most of the generated files. That's great, but what if we are using the generated code into a pipeline that generates an artifact? Probably we're going to have issues because the import statement (with the typo) will fail or similar.

About mustache limitations, I agree, I'll try to work on it to understand how complex would be to generate "elegant" enough code.

Reply:
Keeping this open as we're going to move the effort to address this specific comment into a separate PR